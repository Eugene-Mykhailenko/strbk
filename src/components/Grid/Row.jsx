import React from 'react';
import PropTypes from 'prop-types';
import styles from './Grid.scss';

const Row = (props) => {
  const { children } = props;

  return <div className={styles.row}>{children}</div>;
};

Row.propTypes = {
  /** Special property to pass children
   * elements directly into component output. */
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Row;
