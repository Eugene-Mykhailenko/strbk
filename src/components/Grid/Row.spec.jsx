import React from 'react';
import Row from './Row';

const div = <div className="find-me">Lorem Ipsum is simply dummy text</div>;

describe('Row', () => {
  it('should render with no children by default', () => {
    const row = shallow(<Row>{div}</Row>);

    expect(row.prop('className')).toEqual('row');
    expect(row).toMatchSnapshot();
  });

  it('should render provided children', () => {
    const row = shallow(<Row>{div}</Row>);

    expect(row.find('.find-me')).toHaveLength(1);
    expect(row).toMatchSnapshot();
  });
});
