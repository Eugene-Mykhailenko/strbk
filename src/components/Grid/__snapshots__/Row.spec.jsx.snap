// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row should render provided children 1`] = `
<div
  className="row"
>
  <div
    className="find-me"
  >
    Lorem Ipsum is simply dummy text
  </div>
</div>
`;

exports[`Row should render with no children by default 1`] = `
<div
  className="row"
>
  <div
    className="find-me"
  >
    Lorem Ipsum is simply dummy text
  </div>
</div>
`;
